lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3SecurityMappingProvider.java

```diff
@@ -131,6 +136,31 @@ private Optional<String> getKmsKeyIdFromExtraCredential(ConnectorIdentity identi
         return kmsKeyIdCredentialName.map(name -> identity.getExtraCredentials().get(name));
     }
 
+    private Optional<String> selectCustomerKey(S3SecurityMapping mapping, ConnectorIdentity identity)
+    {
+        Optional<String> userSelected = getCustomerKeyFromExtraCredential(identity);
+
+        if (userSelected.isEmpty()) {
+            return mapping.customerKey();
+        }
+
+        String selected = userSelected.get();
+
+        // selected Customer key must match default or be allowed
+        if (!selected.equals(mapping.customerKey().orElse(null)) &&
+                !mapping.allowedCustomerKeys().contains(selected) &&
```

Comment:
We're missing docs that would explain the relationship between `customerKey` and `allowedCustomerKeys`. `customerKey` is optional, but if users only set `customerKey` without any `allowedCustomerKeys`, this will always silently fail. Maybe we should throw an exception in this case?

Reply:
Thanks! But we're still missing docs that would describe it. KMS is well documented.

Reply:
added! thanks